src/worldfactory.cpp

```diff
@@ -144,21 +144,22 @@ WORLDPTR worldfactory::make_new_world( bool show_prompt )
         // set up window
         catacurses::window wf_win = catacurses::newwin(FULL_SCREEN_HEIGHT, FULL_SCREEN_WIDTH, iOffsetY, iOffsetX);
 
-        int curtab = 0;
-        int lasttab; // give placement memory to menus, sorta.
-        const int numtabs = tabs.size();
-        while (curtab >= 0 && curtab < numtabs) {
+        size_t curtab = 0;
+        size_t lasttab; // give placement memory to menus, sorta.
+        const size_t numtabs = tabs.size();
+        while ((unsigned long)curtab < (unsigned long)numtabs) {
             lasttab = curtab;
             draw_worldgen_tabs(wf_win, curtab);
             curtab += tabs[curtab](wf_win, retworld);
 
-            if (curtab < 0) {
+            // If it is -1, or for unsigned size_t, it would be max.
+            if (curtab == (size_t)-1) {
```

Comment:
Please take another approach to addressing this, casting a negative number to an unsigned number and relying on unsigned underflow behavior is far more confusing than having a few warnings floating around.